proxy/src/managers/in_mem/block_list.rs

```diff
@@ -19,7 +19,16 @@ impl BlockList for InMemoryBlockList {
         }
     }
 
-    fn get_list_for_user(&mut self, account_id: &AccountId) -> Option<&Vec<AccountId>> {
-        self.block_list.get(account_id)
+    fn check_for_blocked_user(
+        &mut self,
+        user_account_id: &AccountId,
+        blocked_account_id: &AccountId,
+    ) -> bool {
+        if let Some(list) = self.block_list.get(user_account_id) {
+            if list.contains(blocked_account_id) {
+                return true;
+            }
+        }
+        false
```

Comment:
```suggestion
     self.block_list.entry(user_account_id).or_insert_with(||Vec::new()).contains(blocked_account_id)
```

please check this before commiting